src/HaskellCI.hs

```diff
@@ -761,18 +743,25 @@ genTravisFromConfigs argv opts isCabalProject config prj@Project { prjPackages =
         tellStrLns
             [ comment "Constraint sets"
             , sh "rm -rf cabal.project.local"
-            , ""
             ]
         forM_ constraintSets $ \cs -> do
             let name = csName cs
-            let constraintFlags = concatMap (\x ->  " --constraint='" ++ x ++ "'") (csConstraints cs)
-            let cmd | csRunTests cs = "${CABAL} new-test  -w ${HC} --enable-tests  --enable-benchmarks"
-                    | otherwise     = "${CABAL} new-build -w ${HC} --disable-tests --disable-benchmarks"
             tellStrLns [ comment $ "Constraint set " ++ name ]
+
+            let shForCs = shForJob versions' (csGhcVersions cs)
+            let testFlag = if csTests cs then " --enable-tests" else " --disable-tests"
```

Comment:
The space character in `" --enable-tests"` is unnecessary, since this ends up being fed into `unwords`, which adds the necessary spacing anyways. (Similarly for `" --disable-tests"`, `" --enable-benchmarks"`, and `" --disable-benchmarks"`.)

Reply:
good catch.

Reply:
it's interesting that per package `tests: False` overrides cli argument `--enable-tests` btw. I find that slightly weird.